src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -160,6 +160,72 @@ abstract class RefChecks extends Transform {
         })
       }
     }
+    private def checkDubiousOverloads(clazz: Symbol): Unit = if (settings.warnDubiousOverload) {
+      // nilary members or methods with leading implicit params
+      def ofInterest(tp: Type): Boolean = tp match {
+        case mt: MethodType => mt.isImplicit
+        case PolyType(_, rt) => ofInterest(rt)
+        case _: NullaryMethodType | _ => true
```

Comment:
```suggestion
        case _ => true
```

`_: NullaryMethodType | _` could be considered clarification or confusing, depending on the time of day..

Reply:
Normally I'd put it in `/* _: NullaryMethodType |*/` and then a reviewer would say `delete?`. I guess it could go in a regular line comment. This is an example where one forgets the model (that there is a nullary MT) if one has been away from it. Why doesn't the compiler help? I don't want to enumerate all the types I'm not interested in.